scrapper.py

```diff
@@ -26,38 +39,135 @@ class Crawler:
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(self.article_url, self.article_id)
+
+    def parse(self):
+        """
+        Extracts all necessary data from the article web page
+        """
+        response = requests.get(self.article_url, HEADERS)
+        article_bs = BeautifulSoup(response.text, 'html.parser')
+
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+    def _fill_article_with_text(self, article_bs):
+        """
+        Fills the Article instance with text
+        """
+        fulltext = article_bs.find('div', {'class': 'fulltext'})
+        page_link = fulltext.find('a')['href']  # link to a page with pdf
+
+        response_pdf = requests.get(page_link, HEADERS)  # downloads page with pdf to find a link for download pdf
+        pdf_bs = BeautifulSoup(response_pdf.text, 'html.parser')
+
+        container = pdf_bs.find('div', id='pdfDownloadLinkContainer')
+        download_link = container.find('a')['href']
+
+        pdf = PDFRawFile(download_link, self.article_id)
+
+        pdf.download()
+        self.article.text = pdf.get_text()
+
+    def _fill_article_with_meta_information(self, article_bs):
+        """
+        Fills the Article instance with meta information
+        """
+        article_title = article_bs.find('meta', {'name': 'description'})['content']
+        self.article.title = article_title
+
+        article_author = article_bs.find('meta', {'name': 'DC.Creator.PersonalName'})
+        if not article_author:
+            article_author = 'NOT FOUND'
+        else:
+            article_author = article_author['content']
+        self.article.author = article_author
+
+        date_raw = article_bs.find('meta', {'name': 'DC.Date.dateSubmitted'})['content']
+        article_date = datetime.datetime.strptime(date_raw, '%Y-%m-%d')
+        self.article.date = article_date
 
 
 def prepare_environment(base_path):
     """
     Creates ASSETS_PATH folder if not created and removes existing folder
     """
-    pass
+    if base_path.exists():
+        shutil.rmtree(base_path)
+    base_path.mkdir(exist_ok=True, parents=True)
 
 
 def validate_config(crawler_path):
     """
     Validates given config
     """
-    pass
+    with open(crawler_path, 'r', encoding='utf-8') as file:
+        config = json.load(file)  # load is for decoding
+
+    seed_urls = config['seed_urls']
+    max_articles = config['total_articles_to_find_and_parse']
+
+    if not seed_urls:
+        raise IncorrectURLError
+
+    for url in seed_urls:
+        if not re.match(r'https?://', url):  # https or http
+            raise IncorrectURLError
+
+    if not isinstance(max_articles, int) or max_articles <= 0:
+        raise IncorrectNumberOfArticlesError
+
+    if max_articles > 200:
+        raise NumberOfArticlesOutOfRangeError
+
+    prepare_environment(ASSETS_PATH)
```

Comment:
this should not happen during the *validation* of config. recall the SRP (single responsibility principle we were talking about during the last lecture?)